codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -160,11 +160,16 @@ public V getAndRemove(K name, V defaultValue) {
     public List<V> getAll(K name) {
         checkNotNull(name, "name");
 
-        LinkedList<V> values = new LinkedList<V>();
-
         int h = hashingStrategy.hashCode(name);
         int i = index(h);
         HeaderEntry<K, V> e = entries[i];
+
+        if (e == null) {
+            return Collections.emptyList();
+        }
+
+        LinkedList<V> values = new LinkedList<V>();
+
         while (e != null) {
```

Comment:
@anuraaga nit: as you have a null check above you can change this to `do {....} while (...)` loop.

Reply:
Yup fixed